### PR TITLE
added testing tools

### PR DESCRIPTION
**Pull request by Moshe-Immerman**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -788,9 +790,11 @@ A curated list of awesome Java frameworks, libraries and software.
 * [pojo-tester](http://www.pojo.pl/) - Automatically performs tests on basic POJO-methods.
 * [PowerMock](https://github.com/jayway/powermock) - Enables mocking of static methods, constructors, final classes and methods, private methods and removal of static initializers.
 * [raml-tester](https://github.com/nidi3/raml-tester) - Tests if a request/response matches a given RAML definition.
+* [Randomized Testing](https://github.com/randomizedtesting/randomizedtesting) - JUnit test runner and plugins for running JUnit tests with pseudo-randomness.
 * [REST Assured](https://github.com/jayway/rest-assured) - Java DSL for easy testing for REST/HTTP services.
 * [Selenide](http://selenide.org/) - Concise API around Selenium to write stable and readable UI tests.
 * [Selenium](http://docs.seleniumhq.org/) - Portable software testing framework for web applications.
+* [Spock](https://github.com/spockframework/spock) - Spock is a developer testing and specification framework for Java and Groovy applications.
```
Comment (akullpp) [inline README.md]: Spock is Groovy not Java, any strong arguments for re-adding it? Since it was deleted with #496

Comment (Moshe-Immerman) [inline README.md]: While spock may have been written in Groovy and support Groovy as a first class citizen it is designed for testing Java application including integration with awesome Java frameworks like Spring, Tapestry, Unitils, etc

Comment (jbduncan) [inline README.md]: If it is decided to reintroduce Spock, then it would be best to just revert the commit which deleted it in the first place.

Comment (akullpp) [inline README.md]: Kind of the same argument could have been made for Swagger...

**Comment by akullpp** [comment]
I will allow it since I'm biased towards Spock as testing framework.

**Pull request merged**